### PR TITLE
Simplify mobile ENV variables

### PR DESCRIPTION
**Pull request by matejkriz**
## Description

- There was a bug using `.env.production` even on develop build of Android in createBundleReleaseJsAndAssets step of gradle.
- ENV variables from `eas.json` are used for EAS builds.
- `.env.development` and `.env.test` files are used for local builds and CI E2E builds.
- Local ENV vars can be overriden by `.env.development.local` and `.env.test.local` files.
- We don't use special decorators any more, so we can switch to "default" babel-preset-expo.

## Related Issue

I hope it can unblock testing of https://github.com/trezor/trezor-suite/issues/13470

### CODE REVIEW

**Review by vdovhanych — APPROVED** [review]

File: suite-native/app/.env.debug [hunk 1]
```diff
@@ -1,3 +0,0 @@
-EXPO_PUBLIC_ENVIRONMENT=debug
-EXPO_PUBLIC_CODESIGN_BUILD=false
-EXPO_PUBLIC_IS_ANALYTICS_LOGGER_ENABLED=false
```
Comment (vytick) [inline suite-native/app/.env.debug:3]: `EXPO_PUBLIC_IS_ANALYTICS_LOGGER_ENABLED`  is now intended to use only in my local .env right?

File: suite-native/config/README.md [hunk 2]
```diff
@@ -1,11 +1,3 @@
 # @suite-native/config
 
 This package contains helper functions based on configuration variables for different environments of our React Native mobile application.
-
-## Config variables
```
Comment (vytick) [inline suite-native/config/README.md:5]: It might be good to add some comment on how to set/change ENV variables somewhere, wdyt?

**Comment by Nodonisko** [comment]
What about e2e where we are triggering JS bundling manually and not in EAS?

**Review by vytick — APPROVED** [review]
Looks good, tested locally in emulator and on device with assembleRelease debug build

File: suite-native/app/.env.debug (hunk 1 above)
Comment (matejkriz) [reply suite-native/app/.env.debug:3]: can be used in .env.development.local, mentioned in README

File: suite-native/config/README.md (hunk 2 above)
Comment (matejkriz) [reply suite-native/config/README.md:5]: Check the README now, please

**Pull request merged**